crates/host/src/wasmbus/mod.rs

```diff
@@ -627,30 +446,41 @@ impl Bus for Handler {
                         invocation.target.public_key
                     ),
                 };
+
+                let timeout = chunked.then_some(CHUNK_RPC_EXTRA_TIME); // TODO: add rpc_nats timeout
+                let request = Request::new().payload(payload.into()).timeout(timeout);
                 let res = nats
-                    .request(topic, request.into())
+                    .send_request(topic, request)
                     .await
                     .context("failed to call provider")
                     .map_err(|e| e.to_string())?;
+
                 let InvocationResponse {
                     invocation_id,
-                    msg,
+                    mut msg,
                     content_length,
                     error,
+                    ..
                 } = rmp_serde::from_slice(&res.payload)
                     .context("failed to decode invocation response")
                     .map_err(|e| e.to_string())?;
                 if invocation_id != invocation.id {
                     return Err("invocation ID mismatch".into());
                 }
-                if let Some(content_length) = content_length {
-                    let content_length = usize::try_from(content_length)
-                        .context("content length does not fit in usize")
+
+                let resp_length = usize::try_from(content_length.unwrap_or_default())
```

Comment:
content_length unwrapped to a 0 is really confusing

Reply:
See above response and let me know whether that resolves this